### ISSUE TITLE
Manual order - error 500 in ajax. Missing products in search box

### ISSUE THREAD

**Issue by Triloworld** [PrestaShop/PrestaShop#32950]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Manual order of some product isn't possible due to missing entries in search.

Bug is show as in this forum with more people, screenshoot and examples: https://www.prestashop.com/forums/topic/1041456-searching-for-products-on-create-order-page/

It is only one product.

### Expected behavior

All product show in search box after typing 5 letter of exact name

### Steps to reproduce

1. Go to Orders (/index.php/sell/orders/)
2. Click to create manual order (/index.php/sell/orders/new)
3. Search for product that exist but not visible (we check: it isn't inventory, not a stock, not price, not visibility and no issue when buying in front or search for clients of frontend)
4. Bar shown that "Is searching" indefinitely
5. On console you get error 500 with message:
"Type error: Argument 2 passed to PrestaShop\PrestaShop\Core\Domain\Product\QueryResult\ProductCombination::__construct() must be of the type string, null given, called in /src/Adapter/Product/QueryHandler/SearchProductsHandler.php on line 273

[Symfony\Component\Debug\Exception\FatalThrowableError 0]"

### PrestaShop version(s) where the bug happened

1.7.8.7, 1.7.7.2

### PHP version(s) where the bug happened

7.3.27-9+0~20210227.82+debian10~1.gbpa4a3d6 and other too

### If your bug is related to a module, specify its name and its version

PrestaShop\PrestaShop\Core\Domain\Product\QueryResult\ProductCombination

### Your company or customer's name goes here (if applicable).

Webo.agency

**Comment by Triloworld** [comment]
Next issue, related in same spot and context:

> Type error: Argument 1 passed to PrestaShop\PrestaShop\Core\Domain\Product\Query\SearchProducts::__construct() must be of the type string, null given, called in /src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php on line 1744

Code:
```php
            $defaultCurrencyId = (int) $this->get('prestashop.adapter.legacy.configuration')->get('PS_CURRENCY_DEFAULT');

            $searchPhrase = $request->query->get('search_phrase');
            $currencyId = $request->query->get('currency_id');
            $currencyIsoCode = $currencyId !== null
                ? Currency::getIsoCodeById((int) $currencyId)
                : Currency::getIsoCodeById($defaultCurrencyId);
            $orderId = null;
            if ($request->query->has('order_id')) {
                $orderId = (int) $request->query->get('order_id');
            }

            /** @var FoundProduct[] $foundProducts */
            $foundProducts = $this->getQueryBus()->handle(new SearchProducts($searchPhrase, 10, $currencyIsoCode, $orderId)); //line 
```

**Comment by Triloworld** [comment]
Error: 

> Type error: Argument 2 passed to PrestaShop\PrestaShop\Core\Domain\Product\QueryResult\ProductCombination::__construct() must be of the type string, null given, called in /src/Adapter/Product/QueryHandler/SearchProductsHandler.php on line 273

Code that refer:
```php
                $productCombination = new ProductCombination(
                    $productAttributeId,
                    $attribute, //This line
                    $combination['quantity'],
                    $this->contextLocale->formatPrice($priceTaxExcluded, $currencyIsoCode),
                    $priceTaxExcluded,
                    $priceTaxIncluded,
                    $combination['location'],
                    $combination['reference'] 
                );

```

**Comment by Triloworld** [comment]
Product with Combinations have NULL attributes (-,- in title). There is something with generating product combinations and their attributes

**Comment by Triloworld** [comment]
Temporary resolve: Recreating product combinations that have NULL atributes
Patch need to be created that omit or block null value. Need some attention from core team and other member to look on that issue

**Comment by florine2623** [comment]
Hello @Triloworld ,

I'm not sure how to create the product from your step 3 : Search for product that exist but not visible (we check: it isn't inventory, not a stock, not price, not visibility and no issue when buying in front or search for clients of frontend)

Could you specify please ?
Thanks!

**Comment by Triloworld** [comment]
> Hello @Triloworld ,
> 
> I'm not sure how to create the product from your step 3 : Search for product that exist but not visible (we check: it isn't inventory, not a stock, not price, not visibility and no issue when buying in front or search for clients of frontend)
> 
> Could you specify please ? Thanks!

Product Combinations that have NULL attributes block that search
Re-creating combinations will only temporary fix issue.

Issue: Something allow Product Combinations without any attributes and when it happen then this search box is break (when manual searching for products to do order). Screens are in link in forum

**Comment by florine2623** [comment]
You should not be having products with NULL combination from the start.

I tried to reproduce your issue with PS 1.7.8.9, 8.0.4 and develop branch. It all seems to be working.
I tested with the following combinations:
<img width="1173" alt="Screenshot 2023-06-20 at 14 43 58" src="https://github.com/PrestaShop/PrestaShop/assets/16019289/03ec02b2-f7bb-410c-8c39-96c4e86d5e41">

It works as expected:

https://github.com/PrestaShop/PrestaShop/assets/16019289/25dadb98-444d-4e64-a5a9-a31ccc9b446e

As you issue is fixed. I will close it.

**Comment by Triloworld** [comment]
Add this issue as reference to ones that have NULL attributes so it will be fixed in future in some other patch where they will find why NULL product combinations was added to catalog.

**Comment by Triloworld** [comment]
> You should not be having products with NULL combination from the start.
> 
> I tried to reproduce your issue with PS 1.7.8.9, 8.0.4 and develop branch. It all seems to be working. I tested with the following combinations: <img alt="Screenshot 2023-06-20 at 14 43 58" width="1173" src="https://user-images.githubusercontent.com/16019289/247118365-03ec02b2-f7bb-410c-8c39-96c4e86d5e41.png">
> 
> It works as expected:
> 
>  Screen.Recording.2023-06-20.at.14.43.37.mov 
> As you issue is fixed. I will close it.

Combination with NULL attributes is what this issue about. You provided with ones that are checked and fine. Add in database so you can recreate issue. I don't have steps to do it in BO
